kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,272 @@
+
+    public KroxyliciousOperatorBundleInstaller(String namespaceInstallTo) {
+        this.namespaceInstallTo = namespaceInstallTo;
+        this.replicas = 1;
+        this.extensionContext = ResourceManager.getTestContext();
+        this.kroxyliciousOperatorName = Constants.KO_DEPLOYMENT_NAME;
+    }
+
+    private static List<File> getOperatorFiles() {
+        if (operatorFiles == null || operatorFiles.isEmpty()) {
+            operatorFiles = Arrays.stream(Objects.requireNonNull(new File(Constants.PATH_TO_OPERATOR_INSTALL_FILES).listFiles())).sorted()
+                    .filter(File::isFile)
+                    .toList();
+        }
+        return operatorFiles;
+    }
+
+    private static List<File> getFilteredOperatorFiles(Predicate<File> predicate) {
+        return getOperatorFiles().stream().filter(predicate).toList();
+    }
+
+    private void applyClusterOperatorInstallFiles(String namespaceName) {
+        for (File operatorFile : getFilteredOperatorFiles(installFiles)) {
+            final String resourceType = operatorFile.getName().split("\\.")[1];
+
+            if (resourceType.equals(Constants.NAMESPACE)) {
+                Namespace namespace = TestFrameUtils.configFromYaml(operatorFile, Namespace.class);
+                if (!NamespaceUtils.isNamespaceCreated(namespace.getMetadata().getName())) {
+                    kubeClient().getClient().resource(namespace).create();
+                }
+            }
+            else {
+                try {
+                    kubeClient().getClient().load(new FileInputStream(operatorFile.getAbsolutePath()))
+                            .inNamespace(namespaceName)
+                            .create();
+                }
+                catch (FileNotFoundException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     */
+    public void prepareEnvForOperator(String clientNamespace) {
+        try {
+            applyCrds();
+        }
+        catch (FileNotFoundException e) {
+            throw new UncheckedIOException(e);
+        }
+        applyClusterOperatorInstallFiles(clientNamespace);
+        applyDeploymentFile();
+    }
+
+    private void applyDeploymentFile() {
+        Deployment operatorDeployment = TestFrameUtils.configFromYaml(getFilteredOperatorFiles(deploymentFiles).get(0), Deployment.class);
+
+        String deploymentImage = operatorDeployment
+                .getSpec()
+                .getTemplate()
+                .getSpec()
+                .getContainers()
+                .get(0)
+                .getImage();
+
+        operatorDeployment = new DeploymentBuilder(operatorDeployment)
```

Comment:
I think this reads much better now.   You can tell what is being overridden at a glance.  Thank you.